src/software/ai/hl/stp/play/dynamic_plays/dynamic_play.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "software/ai/hl/stp/play/play.h"
+
+#include "software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_scorer.h"
+#include "software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_candidate.hpp"
+
+/**
+ * A DynamicPlay is a Play that assigns tactics based on scoring functions
+ * whose parameters and outputs are incrementally adjusted over time based 
+ * on the Play's success. This enables the Play to "learn" which tactics are 
+ * most effective for a given gameplay scenario.
+ */
+class DynamicPlay : public Play 
+{
+   public:
+    /**
+     * Creates a new DynamicPlay
+     *
+     * @param ai_config The AI configuration
+     * @param requires_goalie Whether this play requires a goalie
+     */
+    explicit DynamicPlay(TbotsProto::AiConfig ai_config, bool requires_goalie);
+
+   protected:
+    // TODO (#2359): delete once all plays are not coroutines
+    void getNextTactics(TacticCoroutine::push_type &yield, const World &world) override;
+
+    void updateTactics(const PlayUpdate &play_update) override;
+
+    SupportTacticCandidateVector support_tactic_candidates_;
+
+    std::unique_ptr<FeasibilityScorer> support_tactic_feasibility_scorer_;
```

Comment:
Something I thought about (for the future). Right now, we construct a new Play every time but we probably need to persist Plays to persist information about the support_tactic_success_score_ in the SuccessScorer, so we probably need to add onEnter and onExit methods to clean up histories of some scorers